### PR TITLE
Video edit

### PR DESCRIPTION
**Pull request by idsvn**
#SDBELGA-23 edit video
Create api add thumbnail, cutting video.
requirement: install ffmpeg (sudo apt-get install ffmpeg)
https://wiki.sourcefabric.org/display/NR/Video+Editor

### CODE REVIEW

File: apps/video-edit/__init__.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            'thumbnail': {
+                'href': app.media.url_for_media(media_id, mimetype),
+                'media': media_id,
+                'mimetype': mimetype
+            }
+        }
+        return rendition
+
+    def create_temp_media(self, media_id):
+        media_id = app.media.getFilename(media_id)
```
Comment (petrjasek) [inline apps/video-edit/__init__.py]: why using filename here?

File: superdesk/media/media_operations.py [hunk 2]
```diff
@@ -89,6 +89,21 @@ def process_file_from_stream(content, content_type=None):
...
+    content.seek(0)
+    file_type, ext = content_type.split('/')
+    try:
+        metadata = process_file(content, file_type)
+    except OSError:  # error from PIL when image is supposed to be an image but is not.
+        raise SuperdeskApiError.internalError('Failed to process file')
+    file_name = get_file_name(content)
+    content.seek(0)
+    metadata = encode_metadata(metadata)
+    metadata.update({'length': json.dumps(len(content.getvalue()))})
```
Comment (petrjasek) [inline superdesk/media/media_operations.py]: what is that `json.dumps` doing there?

File: apps/video-edit/__init__.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                if thumbnail_add:
+                    rendition = self.thumbnail_add(path_temp_file, thumbnail_add)
+                    res.update(rendition)
+                if video_cut:
+                    mimetype = item['renditions']['original']['mimetype']
+                    rendition = self.video_cut(path_temp_file, mimetype, video_cut['starttime'],
+                                               video_cut['endtime'])
+                    res.update(rendition)
+            finally:
+                os.remove(path_temp_file)
```
Comment (petrjasek) [inline apps/video-edit/__init__.py]: this should be done probably only if `path_temp_file != None`

**Comment by idsvn** [comment]
We close the pull request for waiting design for separate service.

**Pull request closed without merging**